### PR TITLE
Update integration.md

### PR DESCRIPTION
**Pull request by julienp**
Add some minimal docs about integration tests.

### CODE REVIEW

File: docs/architecture/testing/integration.md [hunk 1]
```diff
@@ -1,4 +1,15 @@
 (integration-testing)=
 ## Integration tests
 
+Integration tests use the built binaries for the CLI and language runtimes. We have some helpers that allow us to create test scenarios and invoke CLI commands.
+
+* [ProgramTest](https://github.com/pulumi/pulumi/blob/master/pkg/testing/integration/program.go#L853): this is typically the easiest way to write an integration test. `ProgramTest` imports a Pulumi program from the `testdata` directory, and runs a series of CLI commands against it.
```
Comment (lunaris) [inline docs/architecture/testing/integration.md:6]: There's a shorthand for GH links which you may have seen `gh-file:<repo>#<path>`. So here I think you could write:

```markdown
* [ProgramTest](gh-file:pulumi#pkg/testing/integration/program.go#L853): this is typically the
```

It's not massively shorter but saves a bit of hardcoding and you get a cute GH icon for it if you are so inclined:

image

File: docs/architecture/testing/integration.md [hunk 2]
```diff
@@ -1,4 +1,15 @@
...
+* [Environment](https://github.com/pulumi/pulumi/blob/master/sdk/go/common/testing/environment.go#L42): this is lower level than `ProgramTest`. It takes care of creating a temporary directory to run commands in via `Environment.RunCommand`. Use `Environment.ImportDirectory` to import a test scenario into the environment.
+
 ### How and when to use
+
+To run a single integration test, run the following command
+
+```bash
+go test -tags=all github.com/pulumi/pulumi/tests/integration -run ${MY_TEST_TO_RUN}
```
Comment (lunaris) [inline docs/architecture/testing/integration.md:14]: N00b Go question: does this command work from the top level of the repository or do you have to be in the `tests` subdirectory?

**Review by lunaris — APPROVED** [review]
Yes! I love it! Maybe this is a chance to point out that you should take care of `make build install` etc. if you are running integration tests, since they use whatever is on `PATH` -- I think that's one of our common gotchas, right? 

:purple_heart: :boom: :100:

File: docs/architecture/testing/integration.md (hunk 2 above)
Comment (julienp) [reply docs/architecture/testing/integration.md:14]: Works from the top level, probably requires the `go.work` file to exist for that?

**Comment by julienp** [comment]
> Yes! I love it! Maybe this is a chance to point out that you should take care of `make build install` etc. if you are running integration tests, since they use whatever is on `PATH` -- I think that's one of our common gotchas, right?

Definitely, I added that now. This doc update is very minimal and rough, but I am trying to get in the habit of adding things here. Easy to improve things with time.

**Pull request merged**